test/FsAutoComplete.Tests.Lsp/CodeFixTests/Tests.fs

```diff
@@ -1637,6 +1637,180 @@ let private generateXmlDocumentationTests state =
+        /// <returns></returns>
+        let f x y = x + y
+        """
+
+      testCaseAsync "missing params and returns for nested function with two parameters"
+      <| CodeFix.check
+        server
+        """
+        let f x y =
+          /// <summary>some comment$0</summary>
+          let g a b =
+            a + b
+          x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        let f x y =
+          /// <summary>some comment</summary>
+          /// <param name=""></param>
+          /// <param name=""></param>
+          /// <returns></returns>
+          let g a b =
+            a + b
+          x + y
+        """
+
+      testCaseAsync "missing single parameter for function with two parameters"
+      <| CodeFix.check
+        server
+        """
+        /// <summary>some comment$0</summary>
+        /// <param name="x"></param>
+        /// <returns></returns>
+        let f x y = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        /// <summary>some comment</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        let f x y = x + y
+        """
+
+      testCaseAsync "missing type parameter for function"
+      <| CodeFix.check
+        server
+        """
+        /// <summary>some comment$0</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        let f x y _ = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        /// <summary>some comment</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name=""></param>
+        /// <typeparam name="'a"></typeparam>
+        /// <returns></returns>
+        let f x y _ = x + y
+        """
+
+      testCaseAsync "not applicable for function without summary"
+      <| CodeFix.checkNotApplicable
+        server
+        """
+        /// some comment$0
+        let f x y = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+
+      testCaseAsync "missing returns for use"
+      <| CodeFix.check
+        server
+        """
+        let f a b _ =
+          /// <summary>$0</summary>
+          use r = new System.IO.BinaryReader(null)
+          
+          a + b
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        let f a b _ =
+          /// <summary></summary>
+          /// <returns></returns>
+          use r = new System.IO.BinaryReader(null)
+          
+          a + b
+        """
+
+      testCaseAsync "not applicable for type with summary"
+      <| CodeFix.checkNotApplicable
+        server
+        """
+        /// <summary>some comment$0</summary>
+        type MyClass() =
+          member val Name = "" with get, set
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
```

Comment:
Can you add a test case showing use on a `MyClass` whose primary constructor has parameters?